### PR TITLE
Add Jest testing to components

### LINKED ISSUE #3: Set up Jest testing framework

### LINKED ISSUE #4: Set up tests for button component

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Added Jest tests to check for each component and that it renders without error.

## QA

 -Pull code
- Run `npm run test`
- Type `a` to run all tests
- All three tests should pass like so:

screen shot 2017-10-12 at 13 35 30

Fixes #3, Fixes #4

### CODE REVIEW

**Comment by nottrobin** [comment]
Couldn't we make the tests run on `npm run test` without the need for user input?

File: src/__tests__/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Button from '../components/Button/Button';
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
+  ReactDOM.render(<Button value="Submit" />, div);
```
Comment (anthonydillon) [inline src/__tests__/Button.test.js]: This test has no assertions. Could you assert the generated markup matches what you expect it to return.

File: src/__tests__/Card.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Card from '../components/Card/Card';
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
+  ReactDOM.render(
+    <Card
+      title="title"
+      cardContent="Lorem ipsum dolor sit amet"
```
Comment (anthonydillon) [inline src/__tests__/Card.test.js]: Same here.

**Comment by barrymcgee** [comment]
@nottrobin It's smarter than that - be default will only run tests on files changed since last commit. It's what comes out of the box with [react-create-app](https://github.com/facebookincubator/create-react-app)

screen shot 2017-10-12 at 11 27 42

**Comment by barrymcgee** [comment]
@anthonydillon Added snapshots, ready for re-review.

**Comment by nottrobin** [comment]
@barrymcgee I do get that. But we should make it so `npm run test` is a task that doesn't require interaction. Surely you can pass the `test` command some options so that it simply runs all tests?

**Comment by anthonydillon** [comment]
@nottrobin @barrymcgee when you have agreed then please merge.

**Comment by barrymcgee** [comment]
@nottrobin Passing a flag is not mentioned [in the docs](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#running-tests). I'm also not keen to start hacking create-react-app as the whole point of it to abstract the tooling and concentrate on the app.

**Comment by nottrobin** [comment]
> I'm also not keen to start hacking create-react-app 

This is not a good point. "create" tools are always to create skeleton projects, to help you out until you get to the point where you actually understand the thing. We only use defaults where they're useful to us.

@barrymcgee That wasn't hard: https://github.com/vanilla-framework/vanilla-components/pull/22

I am extremely unhappy that you merged this before agreeing with me. I know I didn't explicitly block the PR, but that is not the spirit of reviews. Merges should always be cautious and you don't merge if anyone might still have objections. Also, @anthonydillon explicitly said "when you have agreed".

It also seems to me that tests in fact don't pass:

``` bash
$ npm run test

> vanilla-framework-react@0.1.0 test /home/robin/git/vanilla-components
> CI=true react-scripts test --env=jsdom

{
  "name": "vanilla-framework-react",
 FAIL  src/components/Button/Button.test.js
  ● Test suite failed to run

    Cannot find module './Button.css' from 'Button.js'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at Object.<anonymous> (src/components/Button/Button.js:3:1)

 FAIL  src/components/Card/Card.test.js
  ● Test suite failed to run

    Cannot find module './Card.css' from 'Card.js'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at Object.<anonymous> (src/components/Card/Card.js:3:1)

 FAIL  src/__tests__/App.test.js
  ● Test suite failed to run
{
Run tests without user input
 FAIL  src/__tests__/App.test.js
  ● Test suite failed to run

    Cannot find module './Button.css' from 'Button.js'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at Object.<anonymous> (src/components/Button/Button.js:3:1)

 FAIL  src/components/Button/Button.test.js
  ● Test suite failed to run

    Cannot find module './Button.css' from 'Button.js'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at Object.<anonymous> (src/components/Button/Button.js:3:1)

 FAIL  src/components/Card/Card.test.js
  ● Test suite failed to run

    Cannot find module './Card.css' from 'Card.js'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at Object.<anonymous> (src/components/Card/Card.js:3:1)

Test Suites: 3 failed, 3 total
Tests:       0 total
Snapshots:   0 total
Time:        1.381s
Ran all test suites.
Ran all test suites.
```

**Pull request merged**